### PR TITLE
bugfix dali.download(), implement dali.Track.to_jams()

### LINKED ISSUE #197: DALI and GTZAN_genre lack a `to_jams` method

### LINKED ISSUE #229: dali._load_metadata() points to the wrong path
Follow-up from a discussion with @gabolsgabs
Here are our instructions on how to download DALI:

```

        To download this dataset, visit:
        https://zenodo.org/record/2577915 and request access.

        Once downloaded, unzip the file DALI_v1.0.zip
        and place the result in:
        /Users/vl238/mir_datasets/DALI/DALI/annotatios

        Use the function dali_code.get_audio you can find at:
        https://github.com/gabolsgabs/DALI for getting the audio and place them at:
        /Users/vl238/mir_datasets/DALI/DALI/audio
```

First, the path is wrong. It should be `/Users/vl238/mir_datasets/DALI/annotatios`, not `/Users/vl238/mir_datasets/DALI/annotations` (note the two `DALI` folders, and the typo in `annotatios`)

The problem is that this Zenodo record does not have metadata. So this will work: (because it's a property)

```
dali.load()['001940b614eb43f4a0c826d49a67d66d'].words
```

This won't work (because it's an attribute)
```
dali.load()['001940b614eb43f4a0c826d49a67d66d'].album
```

Luckily, @gabolsgabs has agreed to host the metadata JSON file on GitHub:
https://github.com/gabolsgabs/DALI/blob/master/code/DALI/files/dali_v1_metadata.json

He release DALI v1.1 with this JSON file. So from the scope of mirdata, we should be able to require DALI v1.1 and load that file automatically in the `mirdata.dali._load_metadata`

@rabitt i suppose this is urgent?

### PR DESCRIPTION
**Pull request by lostanlen**
closes #229 bug
closes #197 `NotImplementedError`

many thanks to @gabolsgabs for this update
triaged for v0.2

TODO:
- [x] implement `dali.Track.to_jams()`
- [x] restore dali module to `test_loaders.py`
- [x] style check
- [x] usability check

### CODE REVIEW

**Comment by lostanlen** [comment]
This is ready for review. Note that i ended up using a `raw.githubusercontent.com` address, not a `github.com` address, because the latter kept giving me inconsistent checksums

**Comment by rabitt** [comment]
Before reviewing - you're implementing `dali.Track.to_jams(duration)`, but we didn't yet decide on #226 . Let's wait for that decision first?

File: mirdata/dali.py [hunk 1]
```diff
@@ -166,8 +173,14 @@ def audio(self):
...
     def to_jams(self):
-        """(Not Implemented) Jams: the track's data in jams format"""
-        raise NotImplementedError
+        """Jams: the track's data in jams format"""
+        # Load metadata
+        metadata = {k: v for k, v in self._track_metadata.items() if v is not None}
+        y, sr = self.audio
+        metadata['duration'] = librosa.get_duration(y=y, sr=sr)
+        return jams_utils.jams_converter(
+            lyrics_data=[(self.words, None)], metadata=metadata
```
Comment (rabitt) [inline mirdata/dali.py:182]: there are multiple levels of lyrics annotations - 
```
lyrics_data = [
    (self.words, 'word-aligned lyrics'),
    (self.lines,  'line-aligned lyrics'),
    (self.paragraphs, 'paragraph-aligned lyrics')
]
```

And we should add the note data too - 
`note_data=[(self.notes, 'annotated vocal notes')]`

File: mirdata/dali.py [hunk 2]
```diff
@@ -184,7 +197,7 @@ def load_audio(audio_path):
     return librosa.load(audio_path, sr=None, mono=True)
 
 
-def download(data_home=None):
+def download(data_home=None, force_overwrite=False):
```
Comment (rabitt) [inline mirdata/dali.py:200]: `force_overwrite` isn't used in this function

**Comment by lostanlen** [comment]
@rabitt i implemented `dali.Track.to_jams()` by looking up the duration of the audio. This is ready for CR

File: mirdata/dali.py [hunk 3]
```diff
@@ -197,6 +210,8 @@ def download(data_home=None):
     if data_home is None:
         data_home = utils.get_default_dataset_path(DATASET_DIR)
 
+    download_utils.downloader(data_home, file_downloads=[METADATA_REMOTE])
+
     print(
```
Comment (rabitt) [inline mirdata/dali.py:215]: this module was written before we added the `info_message` argument to the `downloader` function. While we're here, could you move the print string to `info_message=...`?

File: mirdata/dali.py (hunk 2 above)
Comment (rabitt) [reply mirdata/dali.py:200]: maybe you meant to pass it to `download_utils.downloader`?

File: tests/test_loaders.py [hunk 4]
```diff
@@ -67,7 +68,6 @@ def test_load_and_trackids():
 
 
 def test_track():
-    jams_temporary_exceptions = ['dali']
```
Comment (rabitt) [inline tests/test_loaders.py:70]: ❤️

File: setup.py [hunk 5]
```diff
@@ -48,16 +48,15 @@
                 'testcontainers>=2.3',
                 'future==0.17.1',
                 'coveralls>=1.7.0',
-                'dali-dataset',
             ],
             'docs': [
-                'DALI-dataset>=1.0.0',
+                'dali-dataset==1.1',
```
Comment (rabitt) [inline setup.py:53]: To double check - In line 31, is `import DALI` still correct with the new version of `dali-dataset`?

Comment (lostanlen) [reply setup.py:53]: yes, i just double-checked

File: mirdata/dali.py (hunk 1 above)
Comment (lostanlen) [reply mirdata/dali.py]: thanks!

File: mirdata/dali.py (hunk 2 above)
Comment (lostanlen) [reply mirdata/dali.py:206]: oops, good catch

File: mirdata/dali.py (hunk 3 above)
Comment (lostanlen) [reply mirdata/dali.py]: good idea. done

File: tests/test_loaders.py (hunk 4 above)
Comment (lostanlen) [reply tests/test_loaders.py:70]: `_temporary_` is the operative word here ;)

**Comment by rabitt** [comment]
Feel free to merge, despite the patch coverage issue - it's just because the number of lines of code in dali increased without the tests changing.

**Comment by lostanlen** [comment]
many thanks for the CR @rabitt
merging

**Comment by lostanlen** [comment]
@rabitt huh, i can't merge. i'm on my phone. please merge at your convenience

**Pull request merged**